### ISSUE TITLE
Build fails with: no matching function for call to 'Gtk::CheckMenuItem::set_active(Glib::RefPtr<GtkSpell::Checker>)'

### ISSUE THREAD

**Issue by innir** [manisandro/gImageReader#102]
Hi,

I saw gImageReader fails to build with this error:

```
/build-1st/gimagereader-3.1.91/gtk/src/OutputEditorText.cc: In member function 'void OutputEditorText::completeTextViewMenu(Gtk::Menu*)':
/build-1st/gimagereader-3.1.91/gtk/src/OutputEditorText.cc:204:69: error: no matching function for call to 'Gtk::CheckMenuItem::set_active(Glib::RefPtr<GtkSpell::Checker>)'
  item->set_active(GtkSpell::Checker::get_from_text_view(*m_textView));
                                                                     ^
In file included from /usr/include/gtkmm-3.0/gtkmm.h:127:0,
                 from /build-1st/gimagereader-3.1.91/gtk/src/common.hh:23,
                 from /build-1st/gimagereader-3.1.91/gtk/src/FileDialogs.hh:23,
                 from /build-1st/gimagereader-3.1.91/gtk/src/OutputEditorText.cc:22:
/usr/include/gtkmm-3.0/gtkmm/checkmenuitem.h:124:8: note: candidate: void Gtk::CheckMenuItem::set_active(bool)
   void set_active(bool is_active =  true);
        ^~~~~~~~~~
```

The config is:

```
cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DINTERFACE_TYPE=gtk -DENABLE_VERSIONCHECK=0
-- GSettings schemas will be installed locally.
-- GSettings shemas will be compiled.
-- The C compiler identification is GNU 6.2.0
-- The CXX compiler identification is GNU 6.2.0
[...]
-- Checking for module 'gtkmm-3.0'
--   Found gtkmm-3.0, version 3.22.0
-- Checking for module 'gtksourceviewmm-3.0'
--   Found gtksourceviewmm-3.0, version 3.18.0
-- Checking for module 'gtkspellmm-3.0>=3.0.4'
--   Found gtkspellmm-3.0, version 3.0.4
-- Checking for module 'cairomm-1.0'
--   Found cairomm-1.0, version 1.12.0
-- Checking for module 'pangomm-1.4'
--   Found pangomm-1.4, version 2.40.1
-- Checking for module 'poppler-glib'
--   Found poppler-glib, version 0.44.0
-- Checking for module 'json-glib-1.0'
--   Found json-glib-1.0, version 1.2.2
-- Checking for module 'libxml++-2.6'
--   Found libxml++-2.6, version 2.40.1
-- GSettings schemas will be installed into /usr/share/glib-2.0/schemas/
-- Configuring done
-- Generating done
```

So it might be gcc 6.2 and/or GTK+ 3.22 related.

You can find the full build log at: https://tests.reproducible-builds.org/debian/rbuild/unstable/amd64/gimagereader_3.1.91-1.rbuild.log

Best,
Philip

**Comment by manisandro** [comment]
Hi
Looks like the implicit `bool` conversion isn't performed. Does

    item->set_active(GtkSpell::Checker::get_from_text_view(*m_textView) != nullptr)

work?

**Comment by innir** [comment]
Thanks for the quick reply, I'll check tonight.

**Comment by innir** [comment]
Oh, you already fixed it in August with commit 779f447ccfec81a6fe301b73fdf5d8c401d12e64
